File: app/src/main/java/com/example/githubrowse/search_user/UserSearchAdapter.java
package com.example.githubrowse.search_user;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.githubrowse.R;
import com.example.githubrowse.model.Items;


import java.util.List;

public class UserSearchAdapter extends RecyclerView.Adapter<UserSearchAdapter.MyViewHolder> {

    private List<Items> items;
    private Context context;

    public UserSearchAdapter(List<Items> items, Context context) {
        this.items = items;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.user_item, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.name.setText(items.get(position).getLogin());
//        Glide.with(context)
//                .load(items.get(position).getAvatar_url())
//                .apply(
//                        RequestOptions.placeholderOf(R.drawable.ic_github)
//                )
//                .transition(DrawableTransitionOptions.withCrossFade())
//                .into(holder.avatar);

        Glide.with(context).load(items.get(position).getAvatar_url()).into(holder.avatar);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {
        TextView name;
        ImageView avatar;
        MyViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
            avatar = itemView.findViewById(R.id.avatar);
        }
    }
}

File: app/src/main/java/com/example/githubrowse/search_user/UserSearchAdapter.kt
package com.example.githubrowse.search_user

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.bumptech.glide.Glide
import com.example.githubrowse.R
import com.example.githubrowse.model.Items
import com.example.githubrowse.model.Owner
import com.example.githubrowse.model.Repo

class UserSearchAdapter(private val items: List<Owner>, private val context: Context) : RecyclerView.Adapter<UserSearchAdapter.MyViewHolder>() {

    private lateinit var onItemClickedListener: OnItemClickedListener

    interface OnItemClickedListener {

        fun onClicked(repo: Owner?, pos: Int)
    }

    fun setOnItemClickedListener(listener: OnItemClickedListener) {
        this.onItemClickedListener = listener
    }





    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.user_item, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.name.text = items[position].login
        holder.itemView.setOnClickListener{v->


            onItemClickedListener.onClicked(items[position],position)
        }

        Glide.with(context).load(items[position].avatar_url).into(holder.avatar)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name: TextView
        var avatar: ImageView

        init {
            name = itemView.findViewById(R.id.name)
            avatar = itemView.findViewById(R.id.avatar)
        }
    }
}
